application/worker.py

```diff
@@ -105,6 +106,7 @@ def execute(self, directory, id):
             session.add(validation_task)
             session.commit()
             validation_task_id = str(validation_task.id)
+            super(task, self).__setattr__("validation_task_id",validation_task_id)
```

Comment:
Instead of storing `validation_task_id`, why not store `validation_task` directly? Can directly be updated in the code below.

Also, why are you using `super(task, self).__setattr__` instead of just `self.validation_task = validation_task`?

Reply:
I think it's preferable to use `validation_task_id` to retrieve the object already stored in the DB, because this object won't be bound to a session anymore when it's needed to register the times. Or one solution would be to expunge this object https://docs.sqlalchemy.org/en/20/orm/session_state_management.html#expunging.

Reply:
Ok, indeed the session makes this annoying, good point.

We probably also should have done this differently where the interaction with the DB session happens centrally and not in every task.execute() individually.

There's some advise regarding that here as well https://docs.sqlalchemy.org/en/14/orm/session_basics.html#when-do-i-construct-a-session-when-do-i-commit-it-and-when-do-i-close-it which we don't really follow.

One solution I see for now is that everything related to session moves to the `task` supertype directly. Where it initializes a `self.session` member variable (or obtains one from the worker) that keeps alive for as long as the task is active (might be a bad idea in general, but with only one or a handful of workers it should be ok). What do you think. It would save some time I guess.

The thing is, an ORM (sqalchemy) should make our lives easier. By allowing interfaces to database tables as intelligent python objects. The way we are using it with constantly fetching objects by id, eliminates this ease of use.

Reply:
> We probably also should have done this differently where the interaction with the DB session happens centrally and not in every task.execute() individually.

Agree, we could indeed do the following: for each task create the validation_task instance and execute the `run_task` call within the session context https://github.com/AECgeeks/ifc-pipeline-validation/blob/development/application/worker.py#L474. We can set this validation_task instance start and end times before and after `run_task`.